src/graphics/vertex-iterator.js

```diff
@@ -1,39 +1,67 @@
 Object.assign(pc, function () {
     'use strict';
 
-    function VertexIteratorAccessor(buffer, vertexElement) {
+    function VertexIteratorAccessor(buffer, vertexElement, vertexFormat) {
         this.index = 0;
+        this.numComponents = vertexElement.numComponents;
+
+        // map only section of underlaying buffer for non-interleaved format
+        var bufferLength;
+        if (!vertexFormat.interleaved)
+            bufferLength = vertexFormat.vertexCount * vertexElement.numComponents;
 
         switch (vertexElement.dataType) {
             case pc.TYPE_INT8:
-                this.array = new Int8Array(buffer, vertexElement.offset);
+                this.array = new Int8Array(buffer, vertexElement.offset, bufferLength);
                 break;
             case pc.TYPE_UINT8:
-                this.array = new Uint8Array(buffer, vertexElement.offset);
+                this.array = new Uint8Array(buffer, vertexElement.offset, bufferLength);
                 break;
             case pc.TYPE_INT16:
-                this.array = new Int16Array(buffer, vertexElement.offset);
+                this.array = new Int16Array(buffer, vertexElement.offset, bufferLength);
                 break;
             case pc.TYPE_UINT16:
-                this.array = new Uint16Array(buffer, vertexElement.offset);
+                this.array = new Uint16Array(buffer, vertexElement.offset, bufferLength);
                 break;
             case pc.TYPE_INT32:
-                this.array = new Int32Array(buffer, vertexElement.offset);
+                this.array = new Int32Array(buffer, vertexElement.offset, bufferLength);
                 break;
             case pc.TYPE_UINT32:
-                this.array = new Uint32Array(buffer, vertexElement.offset);
+                this.array = new Uint32Array(buffer, vertexElement.offset, bufferLength);
                 break;
             case pc.TYPE_FLOAT32:
-                this.array = new Float32Array(buffer, vertexElement.offset);
+                this.array = new Float32Array(buffer, vertexElement.offset, bufferLength);
                 break;
         }
 
+        // BYTES_PER_ELEMENT is on the instance and constructor for Chrome, Safari and Firefox, but just the constructor for Opera

```

Comment:
what even is 'opera'?

just kidding

Reply:
yeah not sure, that code was there, I just moved it to init time instead of inside .next for each vertex.